### PR TITLE
Added "files" attribute for "npm publish"

### PR DESCRIPTION
**Pull request by dparikh**
"files" attribute lists all the entities (files/folders) included in `npm publish`

No need to include CHANGELOG.*, README.*, LICENSE, package.json

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,9 +1,8 @@
 {
   "private": true,
-  "name": "Web Activities",
-  "version": "0.1.0",
+  "name": "web-activities",
```
Comment (dparikh) [inline package.json]: This was required to run `npm install` (since name can not have spaces)

File: package.json [hunk 2]
```diff
@@ -22,6 +21,13 @@
     "build": "gulp build",
     "dist": "gulp dist"
   },
+  "files": [
```
Comment (dparikh) [inline package.json]: I don't think, I can add inline comments.
I would prefer to mention here that `npm publish` picks up only these resources to be included in tarball.

File: package.json [hunk 3]
```diff
@@ -1,9 +1,8 @@
 {
-  "private": true,
-  "name": "Web Activities",
-  "version": "0.1.0",
+  "private": false,
```
Comment (dparikh) [inline package.json]: `private: false` will not allow `npm publish`

File: package.json (hunk 2 above)
Comment (dvoytenko) [reply package.json]: There should be license as well, right?
Also, as we learned, there's no value to publish `src`. Instead we just need to put a compilable `activities-es6.js` with `module.exports` and bundled `activities.min.js`.

Comment (dparikh) [reply package.json]: Done.
Had to create `./index.js`, and the gulp task will use this file to generate `./index-es6.js`, which is being referenced in the `main` attribute of package.json

**Comment by dparikh** [comment]
I signed it!

File: index-es6.js [hunk 4]
```diff
@@ -0,0 +1,693 @@
+/**
```
Comment (dvoytenko) [inline index-es6.js]: So, do you think we should put this file in .gitignore? Do we really want to always remember to commit it?

File: index-es6.js [hunk 5]
```diff
@@ -0,0 +1,693 @@
...
+  connectHost() {
+    const host = new ActivityIframeHost(this.win_);
+    return host.connect().then(() => host);
+  }
+}
+
+
+
+/**
+ * This file is used to publish to NPM (https://www.npmjs.com/package/web-activities).
```
Comment (dvoytenko) [inline index-es6.js]: This should go up top as `@fileoverview`.

File: build-system/tasks/activities-to-es6.js [hunk 6]
```diff
@@ -77,15 +77,15 @@ function rollupActivities() {
     js = js.replace(/export \{.*\}\;/, '');
     return js;
   }).then(js => {
-    fs.writeFileSync('dist/activities-es6.js', js);
+    fs.writeFileSync('./index-es6.js', js);
```
[the remark this answers is not in the document]
Comment (dparikh) [reply build-system/tasks/activities-to-es6.js]: Done.

File: build-system/tasks/activities-to-es6.js [hunk 7]
```diff
@@ -43,7 +43,7 @@ function rollupActivities() {
   mkdirSync('dist');
   return exec(
     './node_modules/rollup/bin/rollup' +
-    ' src/activities.js' +
+    ' index.js' +
     ' --f es' +//cjs
     ' --no-treeshake --no-strict' +
```
[the remark this answers is not in the document]
Comment (dparikh) [reply build-system/tasks/activities-to-es6.js]: Done

File: index-es6.js (hunk 5 above)
Comment (dparikh) [reply index-es6.js]: Done

File: index-es6.js (hunk 4 above)
Comment (dparikh) [reply index-es6.js]: I think, it would be helpful to commit the generated file.

File: DEPLOYING.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+NPM registry at: https://www.npmjs.com/package/web-activities
+
+NPM login with credentials to web-activities package is required to publish.
+
+Once changes are made to the core library and have been approved, follow
+these steps to publish latest version to NPM:
+
+* Bump the version number in package.json such as: version: "1.0.20" -> "1.0.21"
+* Run the `'gulp activities-to-es6'` task to generate an intermediate file at
```
Comment (dvoytenko) [inline DEPLOYING.md]: We need to combine `gulp activities-to-es6` with `gulp dist`. We should only ever call one `gulp dist` for something like this. Further, no need to mention the intermediate files, just the end result.

File: package.json [hunk 9]
```diff
@@ -22,6 +22,13 @@
     "build": "gulp build",
     "dist": "gulp dist"
   },
+  "files": [
+    "SUPPORT.md",
+    "CONTRIBUTING.md",
+    "CODE_OF_CONDUCT.md",
+    "LICENSE",
+    "index-es6.js"
```
Comment (dvoytenko) [inline package.json]: Now it's just `activities.js`, right?

Comment (dparikh) [reply package.json]: oops!, Done.

File: DEPLOYING.md (hunk 8 above)
Comment (dparikh) [reply DEPLOYING.md]: Updated the `gulp dist` task to include `activities-to-es6` and updated the doc.

**Pull request merged**